test/test_nn.py

```diff
@@ -16897,6 +16897,49 @@ def test_cross_entropy_loss_one_hot_target(self, device):
                 output_one_hot = m(input, target_one_hot)
                 self.assertEqual(output, output_one_hot)
 
+    def test_cross_entropy_consistent_label_smoothing_errors(self, device):
+        N, C = 10, 4
+        input = torch.randn(N, C, requires_grad=True, device=device)
+        target = torch.empty(N, dtype=torch.long, device=device).random_(0, C)
+
+        bad_target = torch.empty_like(input, device=device)
+        loss = nn.CrossEntropyLoss(label_smoothing=0.3)
+        with self.assertRaisesRegex(RuntimeError,
+                                    r'target must be specified as class indices to use label_smoothing'):
+            loss(input, bad_target)
+
+        loss = nn.CrossEntropyLoss(label_smoothing=0.3, ignore_index=1)
+        with self.assertRaisesRegex(RuntimeError, r'ignore_index is not supported to use label_smoothing'):
+            loss(input, target)
+
+
+    def test_cross_entropy_label_smoothing_consistent_index_target_and_probs(self, device):
```

Comment:
I think it'd be good to also have a set of tests that compare outputs when `label_smoothing > 0.0` is set to a reference of manually applied label smoothing + NLL loss. For reference, the formula for how to smooth labels from the paper is here:
```
class_probs = (1 - eps) * class_probs + eps / num_classes
```